File: 03- March/03- Find the Index of the First Occurrence in a String/03- Find the Index of the First Occurrence in a String (Noura Algohary).py
# Author: Noura Algohary
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        for i in range(len(haystack)):
            if haystack[i] == needle[0]:
                if haystack[i: i+len(needle)] == needle:
                    return i
        
        return -1

File: 03- March/03- Find the Index of the First Occurrence in a String/03- Find the Index of the First Occurrence in a String (Noura Algohary).cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        for(int i = 0;i<haystack.size();i++)
        {
            if(haystack[i] == needle[0])
                if(haystack.substr(i, needle.size()) == needle)
                    return i;
        }

        return -1;
    }
};